### ISSUE TITLE
Full text search.

### ISSUE THREAD

**Issue by jtagcat** [Drive4ik/simple-tab-groups#759]
**Is your feature request related to a problem? Please describe.**
Not finding things.

**Describe the solution you'd like**
While searching, allow dynamic opt-in to full text search within the group, and within all groups.

Keeping it practical:
 - When tab is discarded, unloaded (or archived): Extract text of the tab (whole html is a bit too much).
 - When searching, get text of the active tabs as well. Indexing active tabs beforehand is unpractical for both cpu time, and storage. They change too (much) too often.
 - Depends on #758 in optimizing backups. I can deal with a backup being 1-10G in size fine, but in that case, I would be the only user.

Optional 'keep indexed closed tabs for n days' would be a nice to have as well. Central search for everything possible. 'Find that god damn thing.'

**Describe alternatives you've considered**
Dreaming less of features(, and maybe even implementing them?)

**Comment by jtagcat** [comment]
This could probably better be a seperate add-on interacting with STG (as 'what is within this group, what is not, what is archived, what is cloesd', etc info isn't available).

**Comment by grahamperrin** [comment]
+1 to a separate extension. 

[All Tabs Helper](https://addons.mozilla.org/addon/all-tabs-helper/) is compatible with Firefox Quantum and can seek content in all _loaded_ tabs. 

(I might have tried the feature once or twice, **very** cautiously because I tend to have massive sessions and I wouldn't want any extension to accidentally begin loading non-loaded tabs.)

**Comment by nekohayo** [comment]
See issue #794 for my suggestion of how to make the existing search feature faster; increased performance could solve a portion of your user experience issues, by making you more prone to actually using it as a global search.

**Comment by Drive4ik** [comment]
This search is only possible on open, loaded tabs. If the tab is not loaded, there is no access to its content (html). If the group is archived - the tabs are not physically in the browser, they (URLs) are in the addon memory, that is also no access to the content. 
This is only for new extension. And here we need to think about the feasibility of creating it, as many users will not understand why the extension does not work.

**Comment by jtagcat** [comment]
> If the tab is not loaded, there is no access to its content (html)

It is possible to save it's content before/while archiving. Though, this would have to be implemented in the core extension.
